riff-raff/app/lifecycle/ShutdownWhenInactive.scala

```diff
@@ -1,43 +1,91 @@
 package lifecycle
 
+import com.gu.anghammarad.Anghammarad
+import com.gu.anghammarad.models._
+import conf.Config
 import controllers.Logging
 import deployment.{Deployments, Record}
 import magenta.DefaultSwitch
+import software.amazon.awssdk.regions.internal.util.EC2MetadataUtils
+import software.amazon.awssdk.services.ec2.model.TerminateInstancesRequest
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent._
 
-class ShutdownWhenInactive(deployments: Deployments) extends Lifecycle with Logging {
-  val EXITCODE = 217
+trait WhenInactive extends Lifecycle with Logging {
+  val name: String
+  val description: String
+  val deployments: Deployments
+
+  def action(): Unit
 
   // switch to enable this mode
-  lazy val switch = new DefaultSwitch("shutdown-when-inactive", s"Shutdown riff-raff when there are no running deploys. Turning this on will cause RiffRaff to exit with exitcode $EXITCODE as soon as the last queued deploy finishes.", false) {
-    override def switchOn() = {
+  lazy val switch: DefaultSwitch = new DefaultSwitch(name, description, false) {
+    override def switchOn(): Unit = {
       super.switchOn()
-      // try and shutdown immediately
-      attemptShutdown()
+      attemptAction()
     }
   }
 
-  def attemptShutdown(): Unit = {
+  def attemptAction(): Unit = {
     Future {
-      log.info("Attempting to shutdown: trying to atomically disable deployment")
+      log.info(s"Attempting to $name: trying to atomically disable deployment")
       if (deployments.atomicDisableDeploys) {
         log.info("Deployment disabled, shutting down JVM")
```

Comment:
This log message might be misleading for a rotation/termination action?

Reply:
Good point! Will refactor!